fenix/app/src/main/java/org/mozilla/fenix/gleanplumb/NimbusMessagingStorage.kt

```diff
@@ -78,9 +98,7 @@ class NimbusMessagingStorage(
                         ?: return@mapNotNull null,
                 )
             }.filter {
-                !it.isExpired &&
-                    !it.metadata.dismissed &&
-                    !it.metadata.pressed
+                noFilter || !(it.isExpired || it.metadata.dismissed || it.metadata.pressed)
```

Comment:
The bug here is that the message has expired (i.e. displayed the maximum number of times), so it's unavailable by the time it's needed clicked on. (Regression caused/uncovered by https://github.com/mozilla-mobile/firefox-android/pull/1291)

Is there a simpler fix available; e.g. to make a `createMessage` method used by `getMessages` and a new `getMessage`:

```kotlin
    suspend fun getMessages(): List<Message> {
        val featureValue = messagingFeature.value()
        val nimbusMessages = featureValue.messages

        return nimbusMessages
            .mapNotNull { (key, value) ->
                createMessage(key, value)
            }.filter {
                !it.isExpired &&
                    !it.metadata.dismissed &&
                    !it.metadata.pressed
            }.sortedByDescending {
                it.style.priority
            }
    }

    suspend fun getMessage(key: String): Message? {
        val featureValue = messagingFeature.value()
        val data = featureValue.messages[key] ?: return null
        return createMessage(key, data)
    }
```

Reply:
Good idea, I'll update.  Thanks

Reply:
Couldn't use `createMessage()` since, but I've updated to just use `Message()` directly.